ui/app/AppLayouts/CommunitiesPortal/popups/CreateCommunityPopup.qml

```diff
@@ -145,14 +155,251 @@ StatusStackModal {
+                                    filesList.selectedFiles = selectedFiles
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+
+            StatusBaseText {
+                Layout.fillWidth: true
+                text: qsTr("Please select channel message history files from Discord to import into your new Status Community.")
+                wrapMode: Text.WordWrap
+                visible: filesList.count == 0
+            }
+
+            RowLayout {
+                Layout.alignment: Qt.AlignHCenter
+                spacing: 16
+
+                StatusButton {
+                    text: qsTr("Select files")
+                    onClicked: fileDialog.open()
+                    visible: filesList.count == 0
+                }
+
+                StatusFlatButton {
+                    text: filesList.selectedFiles.length > 0 ? qsTr("Clear items (%1)").arg(filesList.selectedFiles.length) : qsTr("Clear all")
+                    type: StatusBaseButton.Type.Danger
+                    visible: filesList.count > 0
+                    onClicked: {
+                        if (filesList.selectedFiles.length > 0) {
+                            let files = filesList.model
+                            filesList.selectedFiles.forEach(f => {
+                                files.splice(files.indexOf(f), 1)
+                            })
+                            filesList.selectedFiles = []
+                            return filesList.model = files
+                        }
+                        filesList.model = []
+                    }
+                }
+            }
+
+            FileDialog {
+                id: fileDialog
+
+                title: qsTr("Choose files to import")
+                selectMultiple: true
+                folder: shortcuts.pictures
+                nameFilters: [qsTr("JSON files (%1)").arg("*.json")]
+                onAccepted: {
+                    if (fileDialog.fileUrls.length > 0) {
+                        let files = []
+                        files.push(...fileDialog.fileUrls)
+                        filesList.model = files
+                    }
+                }
+            }
+        },
+
+        ColumnLayout {
+
+          id: categoriesAndChannelsView
+
+          readonly property bool canGoNext: root.store.discordChannelsModel.hasSelectedItems
+          property var prevAction: function () {
+            root.store.clearDiscordCategoriesAndChannels()
+            root.store.discordCategoriesAndChannelsExtractionFailed = false
+          }
+
+          Connections {
+              target: root.store
+              onDiscordCategoriesAndChannelsExtractionFailedChanged: {
+                  if (root.store.discordCategoriesAndChannelsExtractionFailed) {
+                    root.currentIndex--
+                    categoriesAndChannelsView.prevAction()
+                  }
+              }
```

Comment:
With this one I need some help from Qt/QML experts. So the idea here is, if we get an "extraction failed" signal, we want to make the stackmodal go one step back and also call the `prevAction()` on the current item (which will in this case reset some model items)

I've experienced one issue here: While this works, if I set the `discordCategoriesAndChannelsExtractionFailed` flag to `false` (so its reset again), the next time I receive a signal which sets the flag back to `true` again, it doesn't trigger the change hook here which causes the stack modal to not go back one step anymore.

Any ideas why that is? 

You can test this by adding `root.store.discordCategoriesAndChannelsExtractionFailed = false` after calling `prevAction()`